link/cgroup.go

```diff
@@ -93,7 +95,17 @@ func newProgAttachCgroup(cgroup *os.File, attach ebpf.AttachType, prog *ebpf.Pro
 		return nil, fmt.Errorf("cgroup: %w", err)
 	}
 
-	return &progAttachCgroup{cgroup, prog, attach, flags}, nil
+	// Retain cgroup and program handles we own to support Update().
+	clone, err := prog.Clone()
+	if err != nil {
+		return nil, err
+	}
+	cgroup, err = os.Open(cgroup.Name())
```

Comment:
Hmm, doesn't this create a TOCTTOU? The name might be pointing at at different cgroup after all. I'd prefer to not reopen the file. If you really want to you could dup the fd, but that seems a lot more cumbersome?

Also creates a hazard since we've already done AttachProgram, aka maybe replaced an existing prog. We can't undo that in this error path. So cloning, etc. must happen before attach I think.

Reply:
Went with a closure that skips Close() for `progAttachCgroup`.